### PR TITLE
Http layer

### PR DESCRIPTION
**Pull request by fabricedesre**
Still incomplete: we send requests but we don't parse the responses yet.

### CODE REVIEW

File: src/simplelink.rs [hunk 1]
```diff
@@ -128,14 +141,29 @@ impl SimpleLink {
...
+        let mac_addr_ptr = if mac_addr_len > 0 {
+            mac_addr.as_ptr()
+        } else {
+            ptr::null() as *const u8
+        };
+        let sec_params_ptr: *const SlSecParams = sec_params.map(|r| &r as *const SlSecParams)
+            .unwrap_or(ptr::null() as *const SlSecParams);
+        let sec_params_ext_ptr: *const SlSecParamsExt =
+            sec_params_ext.map(|r| &r as *const SlSecParamsExt)
+                .unwrap_or(ptr::null() as *const SlSecParamsExt);
```
Comment (dhylands) [inline src/simplelink.rs]: this part will need to be rebased.

File: src/socket_channel.rs [hunk 2]
```diff
@@ -2,42 +2,76 @@
...
+                addr
+            },
+            Err(_) => return Err(ChannelError::InvalidHostName)
+        };
 
+        // TODO: use getsockopt to set the TLS options is tls is true.
+
+        let inaddr = SlSockAddrIn_t {
+            sin_family: Family::AF_INET,
+            sin_port: port,
```
Comment (dhylands) [inline src/socket_channel.rs]: port needs htons too doesn't it?

File: src/socket_channel.rs [hunk 3]
```diff
@@ -51,14 +51,15 @@ impl Channel for SocketChannel {
 
         let inaddr = SlSockAddrIn_t {
             sin_family: Family::AF_INET,
-            sin_port: port,
+            sin_port: unsafe { sl_Htons(port) },
```
Comment (dhylands) [inline src/socket_channel.rs:54]: ok - I see you added that...

**Review by dhylands — APPROVED** [review]
lgtm

File: cc3200-sys/src/socket.rs [hunk 4]
```diff
@@ -0,0 +1,713 @@
...
+        Ok(unsafe { ::core::intrinsics::transmute::<SizeOrError, SocketError>(self) })
+    }
+}
+
+// Enum for the return type of SL_FD_ISSET
+#[repr(i16)]
+#[allow(non_camel_case_types)]
+pub enum SlBoolean {
+    True = 1,
+    Fals = 0,
```
Comment (dhylands) [inline cc3200-sys/src/socket.rs]: Is that a typo? Or is the misspelling deliberate? If deliberate, it's probably worth a comment.

**Review by dhylands — APPROVED** [review]
lgtm - just one minor nit

File: cc3200-sys/src/socket.rs (hunk 4 above)
Comment (fabricedesre) [reply cc3200-sys/src/socket.rs]: That's a typo, good catch!

**Pull request merged**